### ISSUE TITLE
Equivalence between hazardPlot and visreg

### ISSUE THREAD

**Issue by sahirbhatnagar** [sahirbhatnagar/casebase#91]
In trying to create time-dependent hazard ratio plots, I came across the [`visreg`](http://pbreheny.github.io/visreg/) package. 
I found that we can create hazard plots using their package, at least for `glm` objects. Here is an example. I just want this to be documented, because it might also help solve #77. Note that `visreg` uses the offset, so you need to manually set it to 0 in the data attached to the casebase object:

``` r
library(casebase)
#> See example usage at http://sahirbhatnagar.com/casebase/
library(splines)
library(visreg)
data("simdat")
mod_cb <- casebase::fitSmoothHazard(status ~ trt + ns(I(log(eventtime)), df = 3) +
                                        trt:ns(I(log(eventtime)),df=1),
                                    time = "eventtime",
                                    data = simdat,
                                    ratio = 10,
                                    family = "glm")

results0 <- hazardPlot(object = mod_cb, newdata = data.frame(trt = 0),
                       ci.lvl = 0.95, ci = TRUE, lty = 1, line.col = 1, lwd = 2)
head(results0)
#>     trt  eventtime offset predictedloghazard standarderror  lowerbound
#> 1     0 0.03304083      0          -3.772104     0.5365396 0.008036974
#> 1.1   0 0.08321214      0          -3.101928     0.2933333 0.025302540
#> 1.2   0 0.13338344      0          -2.796063     0.2042894 0.040906654
#> 1.3   0 0.18355475      0          -2.604111     0.1595418 0.054106184
#> 1.4   0 0.23372605      0          -2.467233     0.1343677 0.065180896
#> 1.5   0 0.28389735      0          -2.362481     0.1192643 0.074553785
#>     upperbound predictedhazard
#> 1   0.06584142      0.02300360
#> 1.1 0.07989795      0.04496244
#> 1.2 0.09111227      0.06104996
#> 1.3 0.10112333      0.07396890
#> 1.4 0.11037438      0.08481923
#> 1.5 0.11898869      0.09418629
hazardPlot(object = mod_cb, newdata = data.frame(trt = 1), ci = TRUE,
           ci.lvl = 0.95, add = TRUE, lty = 2, line.col = 2, lwd = 2)
legend("topleft", c("trt=0","trt=1"),lty=1:2,col=1:2,bty="y", lwd = 2)
```

![](https://i.imgur.com/S3NNQE2.png)

``` r

mod_cb$data$offset <- 0
visreg(mod_cb,
       xvar = "eventtime",
       by = "trt",
       scale = "linear",
       trans = exp,
       data = mod_cb$data,
       plot = T,
       overlay = TRUE)
```

![](https://i.imgur.com/28EHYf3.png)

<sup>Created on 2020-03-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by sahirbhatnagar** [comment]
This is annoying about `visreg`: it doesn't allow you to supply a different dataset or a subset of the data:

``` r
pacman::p_load(rstpm2)
pacman::p_load(magrittr)
pacman::p_load(visreg)
pacman::p_load(casebase)
brcancer <- transform(brcancer, recyear=rectime / 365.24)
mod_cb <- fitSmoothHazard(censrec ~ hormon + bs(recyear, df = 5), 
                          data = brcancer,
                          time = "recyear")
mod_cb$data$offset <- 0
visreg(mod_cb,
       xvar = "recyear",
       by = "hormon",
       scale = "linear",
       trans = exp,
       data = mod_cb$data,
       plot = T,
       alpha = 1,
       overlay = TRUE)
```

![](https://i.imgur.com/sLquvK1.png)

``` r

visreg(mod_cb,
       xvar = "recyear",
       by = "hormon",
       scale = "linear",
       trans = exp,
       data = mod_cb$data[mod_cb$data$recyear < 4,],
       plot = T,
       alpha = 1,
       overlay = TRUE)
#> Warning in Response(fit, x, trans, alpha, ...): Residuals do not match data;
#> have you changed the original data set? If so, visreg is probably not displaying
#> the residuals for the data set that was actually used to fit the model.
#> Warning in y + rr: longer object length is not a multiple of shorter object
#> length
#> Warning in Response(fit, x, trans, alpha, ...): Residuals do not match data;
#> have you changed the original data set? If so, visreg is probably not displaying
#> the residuals for the data set that was actually used to fit the model.
#> Warning in y + rr: longer object length is not a multiple of shorter object
#> length
#> Error in data.frame(xy[[j]]$x$D, visregRes = xy[[j]]$y$r, visregPos = xy[[j]]$y$pos): arguments imply differing number of rows: 16154, 18503
```

<sup>Created on 2020-03-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by sahirbhatnagar** [comment]
closed by #117
